app/frontend/src/features/auth/useAuthStore.test.ts

```diff
@@ -0,0 +1,368 @@
+    expect(result.current.authState.authenticated).toBe(true);
+  });
+  it("sets error correctly for login fail", async () => {
+    tokenLoginMock.mockRejectedValue({
+      code: StatusCode.PERMISSION_DENIED,
+      message: "Invalid token.",
+    });
+    const { result } = renderHook(() => useAuthStore());
+    expect(result.current.authState.authenticated).toBe(false);
+    await act(() => result.current.authActions.tokenLogin("token"));
+    expect(result.current.authState.authenticated).toBe(false);
+    expect(result.current.authState.error).toBe("Invalid token.");
+  });
+});
+
+describe("signup action", () => {
+  //not testing for the user itself, as that is handled by getUser
+  //and getCurrentUser
+  it("sets authenticated correctly", async () => {
+    signupMock.mockResolvedValue({ jailed: false });
+    getCurrentUserMock.mockResolvedValue(null);
+    const { result } = renderHook(() => useAuthStore());
+    expect(result.current.authState.authenticated).toBe(false);
+    await act(() => result.current.authActions.signup({} as SignupArguments));
+    expect(result.current.authState.authenticated).toBe(true);
+  });
+  it("sets error correctly for login fail", async () => {
+    signupMock.mockRejectedValue({
+      code: StatusCode.PERMISSION_DENIED,
+      message: "Invalid token.",
+    });
+    const { result } = renderHook(() => useAuthStore());
+    expect(result.current.authState.authenticated).toBe(false);
+    await act(() => result.current.authActions.signup({} as SignupArguments));
+    expect(result.current.authState.authenticated).toBe(false);
+    expect(result.current.authState.error).toBe("Invalid token.");
+  });
+});
+
+describe("updateJailStatus action", () => {
+  it("sets jailed to true for jailed user", async () => {
+    getIsJailedMock.mockResolvedValue({ isJailed: true });
+    addDefaultUser();
+    const { result } = renderHook(() => useAuthStore());
+    await act(() => result.current.authActions.updateJailStatus());
+    expect(result.current.authState.jailed).toBe(true);
+    expect(result.current.authState.authenticated).toBe(true);
+  });
+  it("sets jailed to false for non-jailed user", async () => {
+    getIsJailedMock.mockResolvedValue({ isJailed: false });
+    addDefaultUser();
+    const { result } = renderHook(() => useAuthStore());
+    await act(() => result.current.authActions.updateJailStatus());
+    expect(result.current.authState.jailed).toBe(false);
+    expect(result.current.authState.authenticated).toBe(true);
+  });
+  it("throws if not logged in", async () => {
+    logoutMock.mockResolvedValue(new Empty());
+    addDefaultUser();
+    const { result } = renderHook(() => useAuthStore());
+    await act(async () => await result.current.authActions.logout());
+    expect.assertions(1);
+    try {
+      await act(() => result.current.authActions.updateJailStatus());
+    } catch (e) {
+      expect(e).toBeDefined();
```

Comment:
I think it would be useful to check for the actual error message here instead so we know it's not some other random error being thrown.